src/main/kotlin/com/sourcegraph/cody/chat/ui/ContextFilesPanel.kt

```diff
@@ -40,67 +43,148 @@ class ContextFilesPanel(
   }
 
   @RequiresBackgroundThread
-  private fun updateFileList(contextFileNames: Set<String>) {
+  private fun updateFileList(
+      localFiles: Map<LocalContextFile, ContextFile>,
+      remoteFiles: Map<RemoteContextFile, ContextFile>
+  ) {
     val filesAvailableInEditor =
-        contextFileNames
-            .map(Paths::get)
-            .mapNotNull(VirtualFileManager.getInstance()::findFileByNioPath)
+        localFiles
+            .map {
+              val path = Paths.get(it.key.displayText)
+              val findFileByNioPath = VirtualFileManager.getInstance().findFileByNioPath(path)
+              findFileByNioPath ?: return@map null
+              Pair(findFileByNioPath, it.value as? ContextFileFile)
+            }
+            .filterNotNull()
```

Comment:
How is that supposed to work?
You always create `Pair` instance, so `.filterNotNull()` will never filter anything out?

Reply:
Maybe:

```kotlin
           .mapNotNull {
               val path = Paths.get(it.key.displayText)
               val fileByNioPath = VirtualFileManager.getInstance().findFileByNioPath(path)
               val contextFile =  it.value as? ContextFileFile
               if (fileByNioPath != null && contextFile != null) Pair(fileByNioPath, contextFile) else null
             }
```